### PR TITLE
print: Support "supported_output_file_formats" option

### PR DESCRIPTION
**Pull request by GeorgesStavracas**
Respect the allowed output file formats set by apps.

### CODE REVIEW

File: src/print.c [hunk 1]
```diff
@@ -189,6 +189,46 @@ can_preview (void)
...
+
+      formats = g_hash_table_new (g_str_hash, g_str_equal);
+
+      for (size_t i = 0; supported_output_file_formats[i]; i++)
+        g_hash_table_add (formats, (gpointer) supported_output_file_formats[i]);
+
+      /* This is a limitation of GtkPrintCapability - it doesn't have an enum
+       * for SVG files.
+       */
+      if (!g_hash_table_contains (formats, "svg"))
```
Comment (ebassi) [inline src/print.c:220]: Is there a particular reason why you need to put the formats inside a set? We shouldn't be getting duplicates, and if you want to look for a string inside a string vector, you can use [`g_strv_contains()`](https://docs.gtk.org/glib/func.strv_contains.html).

**Review by ebassi — COMMENTED** [review]
Looks good to me, outside of one small nitpick.

File: src/print.c (hunk 1 above)
Comment (ebassi) [reply src/print.c:220]: It's going to be a very small vector, so computationally isn't going to be a huge deal; we probably lose more time creating, populating, and destroying the hash table.

I don't mind using a hash table, just wanted to know if we were supposed to do some validation on the list of supported formats or not.

**Review by ebassi — APPROVED** [review]

**Comment by GeorgesStavracas** [comment]
Thanks!

**Pull request merged**